include/rdma/fabric.h

```diff
@@ -159,6 +159,7 @@ typedef struct fid *fid_t;
 #define FI_PEER_TRANSFER	(1ULL << 36)
 /* #define FI_MR_DMABUF		(1ULL << 40) */
 #define FI_AV_USER_ID		(1ULL << 41)
+#define FI_NO_CONNECT		(1ULL << 42)
```

Comment:
Rather than trying to control the behavior of the implementation, it may make more sense to define some property of the addresses being inserted.  For example, FI_FIREWALL_ADDR can indicate that the addresses being inserted are behind a firewall.  The provider can then decide what to do, or whether it has the ability to access the address behind the firewall.  In the future, there may be some sort of firewall configuration that may allow communicating with the target address, such as going through a proxy.